packages/client/tests/functional/default-selection/tests.ts

```diff
@@ -0,0 +1,86 @@
+import { expectTypeOf } from 'expect-type'
+
+import testMatrix from './_matrix'
+// @ts-ignore
+import type { Prisma as PrismaNamespace, PrismaClient } from './node_modules/@prisma/client'
+
+declare let prisma: PrismaClient
+
+testMatrix.setupTestSuite(({ provider }) => {
+  beforeAll(async () => {
+    const input: Partial<PrismaNamespace.ModelCreateInput> = {
+      value: 'Foo',
+      relation: {
+        create: {},
+      },
+    }
+
+    if (provider !== 'sqlite' && provider !== 'sqlserver') {
+      // @ts-test-if: provider !== 'sqlite' && provider !== 'sqlserver'
+      input.enum = 'A'
+    }
+
+    if (provider === 'postgresql' || provider === 'cockroachdb' || provider === 'mongodb') {
+      // @ts-test-if: provider === 'postgresql' || provider === 'cockroachdb' || provider === 'mongodb'
+      input.list = ['Hello', 'world']
+      // @ts-test-if: provider === 'postgresql' || provider === 'cockroachdb' || provider === 'mongodb'
```

Comment:
would it be worth it to extract this to a variable (e.g. `supportsScalarLists`)?

Reply:
I agree with @aqrln. Let's turn line 23 into something like `if (supportsScalarsList.includes(['postgresql', 'cockroachdb', 'mongodb'])) { /* */ }`

Reply:
Variables won't work for `@ts-test-if` - it is evaluated before test is run using  matrix only. So they'll still be duplicated for type tests. With that in mind, would you still prefer variable at least for runtime conditions?

Reply:
> With that in mind, would you still prefer variable at least for runtime conditions?

Either is fine with me if the runtime condition is duplicated too, but I'd probably prefer it the way it currently is then.